src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -1131,6 +1292,15 @@ query_path(const char *szInput, int *is_target_path, dfs_obj_t **parent, char *i
 		if (atomic_load_relaxed(&daos_inited) == false) {
 			/* daos_init() is expensive to call. We call it only when necessary. */
 
+			/* Check whether daos_init() is running. If yes, pass to the original
+			 * libc functions. Avoid possible daos_init reentrancy/nested call.
+			 */
+			if (!util_bool_compare_and_swap64(&daos_initing, DAOS_INIT_NOT_RUNNING,
```

Comment:
This looks to be using a atomic macro from the (self contained) dav code.

For atomics you should be using either the macros in atomic.h which provide a common interface across gcc/clang and icc or for interception library only (either of them) then the gcc/clang primitives directly.